docs/json_schemas/vault_kv/v0/requirer.json

```diff
@@ -6,10 +6,26 @@
           "description": "Suffix to append to the mount name to get the KV mount.",
           "title": "Mount Suffix",
           "type": "string"
+        },
+        "wrap_ttl": {
+          "anyOf": [
+            {
+              "type": "integer"
+            },
+            {
+              "type": "string"
+            },
+            {
+              "type": "null"
+            }
+          ],
+          "description": "Whether to request approle secret_id as a response-wrapping token with a certain TTL. Can be either an integer number of seconds or a string duration of seconds (`15s`)",
+          "title": "Wrap TTL"
```

Comment:
I don't know that the secret requirer should have the permission to choose their own TTL's on Vault. This seems more like a Vault configuration that should be set in Vault's charmcraft.yaml, just like the other TTL's. 

The Vault charm could provide the wrapping token part of the Vault KV integration still without the requirer having to ask for it.

Reply:
This is actually the [expected behavior](https://developer.hashicorp.com/vault/docs/concepts/response-wrapping#response-wrapping-token-creation) from Vault: "Response wrapping is per-request and is triggered by providing to Vault the desired TTL for a response-wrapping token for that request."

Reply:
As @kayra1  mentioned, this is just how long you, as a client, have to unwrap the value before it is invalidated/forgotten by the server. In other words, Vault will store the value of this token for `wrap_ttl`, and if you haven't requested its value by then, too bad. This limits the attack surface if the token value is never retrieved.

Reply:
@kayra1 and @DanielArndt are correct with what is the usage of `wrap_ttl`. As such, it makes sense to provide it as an option for every integration that requires a wrapping token instead of the secret id, to login with the integration approle.

Reply:
We discussed in a team meeting and we agree that this is a nice feature to have. As @gruyaume is out of office this week, I am taking the liberty to resolve this conversation.